### PR TITLE
Added is_interned method which returns a bool

### PR DESCRIPTION
**Pull request by DanielJoyce**
* Returns true if value already interned.
* Added simple test as well.

### CODE REVIEW

**Comment by DanielJoyce** [comment]
Closes #45

File: src/intern.rs [hunk 1]
```diff
@@ -224,6 +237,15 @@ impl<T: Eq + Hash + Send + Sync + 'static> Intern<T> {
             Intern { pointer: p }
         })
     }
+    /// Check if a value already is interned.
+    ///
+    /// If this value has previously been interned, return true, else returns false
+    pub fn is_interned<'a, Q: ?Sized + Eq + Hash + 'a>(val: &'a Q) -> bool
+    where
+        &'static T: Borrow<Q> + From<&'a Q>,
```
Comment (droundy) [inline src/intern.rs:245]: **suggestion:** You can remove the `From` constraint for this method.

```suggestion
    pub fn is_interned<'a, Q: ?Sized + Eq + Hash + 'a>(val: &'a Q) -> bool
    where
        &'static T: Borrow<Q>,
```

File: src/intern.rs [hunk 2]
```diff
@@ -51,6 +61,9 @@ fn like_doctest_intern() {
     assert_eq!(x, Intern::from_ref("hello"));
     assert_eq!(y, Intern::from_ref("world"));
     assert_eq!(&*x, "hello"); // dereference a Intern like a pointer\
+    let _ = Intern::new("Fortunato");
+    assert!(Intern::is_interned(&"Fortunato"));
+    assert!(!Intern::is_interned(&"Montresor"));
```
Comment (droundy) [inline src/intern.rs:66]: **suggestion:** Could you also test if we can pass just a `&str` to `is_interned`?

```suggestion
    assert!(Intern::is_interned(&"Fortunato"));
    assert!(!Intern::is_interned("Fortunato"));
    assert!(!Intern::is_interned(&"Montresor"));
    assert!(!Intern::is_interned("Montresor"));
```

**Comment by droundy** [comment]
Could you fix the conflicts, and also add an entry to `CHANGELOG.md`?

And thanks for the contribution!

**Comment by DanielJoyce** [comment]
Whoops, will do, sorry, hadn't check back in a while. :)

File: src/intern.rs (hunk 1 above)
Comment (DanielJoyce) [reply src/intern.rs:245]: If I do, then I need to specify generic bounds on line 72:

```
type annotations needed
cannot satisfy `_: std::cmp::Eq`rustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B24%5D?24#file%3A%2F%2F%2Fhome%2Fdanieljoyce%2Fgit%2Fgeneral%2Finternment%2Fsrc%2Fintern.rs)
intern.rs(204, 9): required by a bound in `Intern::<T>::is_interned`
intern.rs(72, 19): consider specifying the generic argument: `::<T>`
```

Comment (DanielJoyce) [reply src/intern.rs:245]: Okay, one sec.

Comment (DanielJoyce) [reply src/intern.rs:245]: If I remove the From<> bounds on T, then it requires dismbiguation per the above error message.

Comment (DanielJoyce) [reply src/intern.rs:245]: If I move the is_interned to the `impl<T: Eq + Hash + Send + Sync + 'static + ?Sized> Intern<T> {` block, then

```
 let _ = Intern::new("Fortunato");
    assert!(Intern::is_interned(&"Fortunato"));
    assert!(Intern::is_interned("Fortunato"));
```

Both check as valid, but when I run the test, ` assert!(Intern::is_interned("Fortunato"));` fails.

**Comment by DanielJoyce** [comment]
Some wonkiness around new() vs from_ref and how I'm checking membership.

 I think the test needs to be a tad more complicated and closer to how the values are inserted.
 
 Depending on which bounds are removed or where the impl for is_interned is moved, some checks fail/pass in different ways.

**Comment by DanielJoyce** [comment]
I think we need to split is_interned and is_interned_ref, 

Also have to be really careful about what Intern<> you choose, as Intern<String> vs Intern<&str> can both be created but don't share membership, so something you think should exist won't because one call to new() retunrs a Intern<String> and the other a Intern<&str> which is likely something you don't want. 

Perhaps the Intern::<_> API should be made explicit like ArenaIntern is? Otherwise if you're not paying attention to the VsCode type hints, you may have created multiple different interners for type vs ref

File: src/intern.rs (hunk 2 above)
Comment (DanielJoyce) [reply src/intern.rs:66]: The problem is the && and & cases can't be unified trivially. Or I don't understand enough rust black magic to do so. 

Basically Intern::is_interned(&"") checks Intern<&str> and Intern::is_interned("") checks Intern<str> which are different.

**Comment by droundy** [comment]
Due to conflict, I merged this outside of github.  I also made some tweaks, so please take a look!

**Pull request closed without merging**